internal/controller/user_controller.go

```diff
@@ -0,0 +1,140 @@
+)
+
+// UserReconciler reconciles a User object
+type UserReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=kuadra.kuadrant.io,resources=users,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=kuadra.kuadrant.io,resources=users/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kuadra.kuadrant.io,resources=users/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the User object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.4/pkg/reconcile
+func (r *UserReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	user := kuadrav1.User{}
+	if err := r.Get(ctx, req.NamespacedName, &user); err != nil {
+		log.Error(err, "unable to fetch User")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	awsAccount := r.createAwsAccountScheme(&user, req.Namespace)
+
+	if err := controllerutil.SetControllerReference(&user, awsAccount, r.Scheme); err != nil {
+		log.Error(err, "Failed to set owner reference for AwsAccount")
+		return reconcile.Result{}, err
+	}
+
+	if existingAwsAccount, err := r.getExistingAwsAccount(ctx, awsAccount); err != nil {
+		if client.IgnoreNotFound(err) == nil {
+			err := r.createAwsAccount(ctx, awsAccount)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		} else {
+			log.Error(err, "Failed to check AwsAccount existence")
+			return reconcile.Result{}, err
+		}
+	} else {
+		err := r.updateAwsAccount(ctx, awsAccount, existingAwsAccount)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *UserReconciler) createAwsAccountScheme(user *kuadrav1.User, namespace string) *kuadrav1.AwsAccount {
+	return &kuadrav1.AwsAccount{
+		TypeMeta: v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      user.Spec.AwsAccount.Spec.User.UserName,
+			Namespace: namespace,
+		},
+		Spec: kuadrav1.AwsAccountSpec{
+			UserName: user.Spec.AwsAccount.Spec.User.UserName,
+			Groups:   user.Spec.AwsAccount.Spec.User.Groups,
+			Zones:    user.Spec.AwsAccount.Spec.User.Zones,
+		},
+	}
+}
+
+// getExistingAwsAccount retrieves the existing AwsAccount object, if it exists.
+func (r *UserReconciler) getExistingAwsAccount(ctx context.Context, awsAccount *kuadrav1.AwsAccount) (*kuadrav1.AwsAccount, error) {
+	existingAwsAccount := &kuadrav1.AwsAccount{}
+	err := r.Get(ctx, types.NamespacedName{Name: awsAccount.Spec.UserName, Namespace: awsAccount.Namespace}, existingAwsAccount)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Just thinking... would it make sense to add `client.IgnoreNotFound(err)` to this return? This might make it easier to avoid the nesting in the `Reconcile` function and improve its readability.